yarn doesn’t work that well 🤷‍♂️

```
$ yarn eslint --fix .
yarn run v1.22.11
$ /Users/tilde/Projects/oss/estree-util-value-to-estree/node_modules/.bin/eslint --fix .
Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db

Why you should do it regularly:
https://github.com/browserslist/browserslist#browsers-data-updating

/Users/tilde/Projects/oss/estree-util-value-to-estree/README.md
  207:7  error  Parsing error: Unexpected token :

/Users/tilde/Projects/oss/estree-util-value-to-estree/src/index.test.ts
  1:1  error  File has too many classes (2). Maximum allowed is 1  max-classes-per-file

/Users/tilde/Projects/oss/estree-util-value-to-estree/src/index.ts
   55:0  error  Missing JSDoc @param "options" description                                                  jsdoc/require-param-description
  176:7  error  Prefer using an optional chain expression instead, as it's more concise and easier to read  @typescript-eslint/prefer-optional-chain
```

---

and editor integrations can often be configured differently that local config files — which is what’s happening for me. I personally don’t think users should have to install other things, unless they’re explicitly documented, and that linting and formatting should also be a script

--- a/README.md
+++ b/README.md
@@ -36,8 +36,11 @@ Currently the following types are supported:
 - URL objects
 - URLSearchParams objects
 
+if `options.jsonFallback` is set to `true`, other values are passed through
+`JSON.parse(JSON.stringify(value))`.
+
 ```js
-import { deepEqual } from 'assert';
+import { deepEqual, throws } from 'assert';
 
 import { valueToEstree } from 'estree-util-value-to-estree';
 
@@ -199,6 +202,41 @@ deepEqual(result, {
     },
   ],
 });
+
+class Point {
+  line: number;
+  column: number;
+  constructor(line: number, column: number) {
+    this.line = line;
+    this.column = column;
+  }
+}
+
+throws(() => valueToEstree(new Point(2, 3)));
+
+deepEqual(valueToEstree(new Point(2, 3), {
+  type: 'ObjectExpression',
+  properties: [
+    {
+      type: 'Property',
+      method: false,
+      shorthand: false,
+      computed: false,
+      kind: 'init',
+      key: { type: 'Literal', value: 'line', raw: '"line"' },
+      value: { type: 'Literal', value: 2, raw: '2' },
+    },
+    {
+      type: 'Property',
+      method: false,
+      shorthand: false,
+      computed: false,
+      kind: 'init',
+      key: { type: 'Literal', value: 'column', raw: '"column"' },
+      value: { type: 'Literal', value: 3, raw: '3' },
+    },
+  ],
+}));
 ```
 
 [codecov badge]:

--- a/src/index.test.ts
+++ b/src/index.test.ts
@@ -453,4 +453,47 @@ describe('valueToEstree', () => {
       new TypeError('Unsupported value: [object Object]'),
     );
   });
+
+  it('should transform to json on unsupported values w/ `jsonFallback`', () => {
+    class Point {
+      line: number;
+      column: number;
+      constructor(line: number, column: number) {
+        this.line = line;
+        this.column = column;
+      }
+    }
+
+    const point = new Point(2, 3);
+
+    // @ts-expect-error This tests an unsupported value.
+    expect(() => valueToEstree(point)).toThrow(
+      new TypeError('Unsupported value: [object Object]'),
+    );
+
+    // @ts-expect-error This tests an unsupported value.
+    expect(valueToEstree(point, { jsonFallback: true })).toStrictEqual({
+      type: 'ObjectExpression',
+      properties: [
+        {
+          type: 'Property',
+          method: false,
+          shorthand: false,
+          computed: false,
+          kind: 'init',
+          key: { type: 'Literal', value: 'line', raw: '"line"' },
+          value: { type: 'Literal', value: 2, raw: '2' },
+        },
+        {
+          type: 'Property',
+          method: false,
+          shorthand: false,
+          computed: false,
+          kind: 'init',
+          key: { type: 'Literal', value: 'column', raw: '"column"' },
+          value: { type: 'Literal', value: 3, raw: '3' },
+        },
+      ],
+    });
+  });
 });

--- a/src/index.ts
+++ b/src/index.ts
@@ -55,7 +55,7 @@ export interface Options {
  * @param options - Additional options to configure the output.
  * @returns The ESTree node.
  */
-export function valueToEstree(value?: Value, options?: Options): Expression {
+export function valueToEstree(value?: unknown, options: Options = {}): Expression {
   if (value === undefined) {
     return { type: 'Identifier', name: 'undefined' };
   }

--- a/src/index.ts
+++ b/src/index.ts
@@ -52,7 +52,7 @@ export interface Options {
  * Convert a value to an ESTree node
  *
  * @param value - The value to convert
- * @param options
+ * @param options - Additional options to configure the output.
  * @returns The ESTree node.
  */
 export function valueToEstree(value?: Value, options?: Options): Expression {

--- a/src/index.ts
+++ b/src/index.ts
@@ -44,13 +44,18 @@ export type Value =
 type ValueSet = Set<Value>;
 type ValueMap = Map<Value, Value>;
 
+export interface Options {
+  jsonFallback?: boolean;
+}
+
 /**
  * Convert a value to an ESTree node
  *
  * @param value - The value to convert
+ * @param options
  * @returns The ESTree node.
  */
-export function valueToEstree(value?: Value): Expression {
+export function valueToEstree(value?: Value, options?: Options): Expression {
   if (value === undefined) {
     return { type: 'Identifier', name: 'undefined' };
   }
@@ -166,5 +171,9 @@ export function valueToEstree(value?: Value): Expression {
       })),
     };
   }
+  if (options && options.jsonFallback) {
+    return valueToEstree(JSON.parse(JSON.stringify(value)));
+  }
+
   throw new TypeError(`Unsupported value: ${String(value)}`);
 }

